src/core/mac/mac_frame.hpp

```diff
@@ -164,33 +180,182 @@ class ExtAddress: public otExtAddress
 } OT_TOOL_PACKED_END;
 
 /**
- * This structure represents an IEEE 802.15.4 Short or Extended Address.
+ * This class represents an IEEE 802.15.4 Short or Extended Address.
  *
  */
-struct Address
+class Address
 {
+public:
     enum
     {
         kAddressStringSize = 18,     ///< Max chars needed for a string representation of address (@sa ToString()).
     };
 
-    uint8_t mLength;                 ///< Length of address in bytes.
-    union
+    /**
+     * This enumeration specifies the IEEE 802.15.4 Address type.
+     *
+     */
+    enum Type
     {
-        ShortAddress mShortAddress;  ///< The IEEE 802.15.4 Short Address.
-        ExtAddress mExtAddress;      ///< The IEEE 802.15.4 Extended Address.
+        kTypeEmpty,              ///< Address is empty (not specified yet).
```

Comment:
nit: "Empty" implies that there should be an address.  IEEE 802.15.4 has valid cases where the address is not present.  Maybe call this "None" instead?

Reply:
Good idea! Will change to `kTypeNone`.